src/libslic3r/GCode/SpiralVase.cpp

```diff
@@ -68,30 +135,69 @@ std::string SpiralVase::process_layer(const std::string &gcode)
                 return;
             } else {
                 float dist_XY = line.dist_XY(reader);
-                if (dist_XY > 0) {
```

Comment:
See my other comments, this check needs to be kept, otherwise we could produce NaN.

Reply:
On it, easy fix. We can't revert this bit to what it was before because of the zero length segments that appear in the gcode but we can definitely avoid doing the computation that explodes and just omit that segment altogether.

Reply:
This should be fixed now! I can't repro so @Noisyfox if you can give this another try or send me repro steps that would be great.